packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -145,12 +145,12 @@ class _DropdownMenuItemButtonState<T> extends State<_DropdownMenuItemButton<T>>
         oldWidget.route.animation != widget.route.animation ||
         oldWidget.route.selectedIndex != widget.route.selectedIndex ||
         widget.route.items.length != oldWidget.route.items.length) {
+      _opacityAnimation.dispose();
```

Comment:
Now if someone adds a call to _setOpacityAnimation they need to remember to dispose the old _opacityAnimation first. That's a slight downgrade, but I think it's still better overall...